### ISSUE TITLE
doorbell videos not downloading

### ISSUE THREAD

**Issue by jg24001** [nayrk/Blink#23]
Just noticed the doorbell videos that I have had for the past month are not in my download folder. The API for the bell camera must be different somehow? I guess I can just have these go to my sync module 2 USB thumbdrive but any other camera I add is fine.

ALso I never thought much of it but the last 6 months or so I get errors downloading thumbnails. I can't scroll back up to see what the error actually is, I'll have to try and screen capture it quickly next time. 

Maybe it is not getting my full list of cameras, or the doorbell is just not accessible on the same server?

**Comment by jg24001** [comment]
Verified that yes, my doorbell camera footage is there in my USB backup, but all the backup filenames are different from the usual clip names, just doorbell123 or frontdoor89 in order which I don't know if it is sequential or not without going through all of them.

I usually run the script with teamviewer when I'm not home, and I can't robotically remotely move the USB thumbdrive in and out of the sync2 module that has the doorbell camera :)

**Comment by jg24001** [comment]
All doorbell files are there on USB drive in chronological order but the file's timestamp which used to also be in the filename, is just whenever the overnight backup occurred. Not ideal but at least we can delete the doorbell cameras off the cloud in the app to clear up space. I have so many cameras thanks to all the amazon sales that it fills up quickly for me, even with several sync modules which give the early adopters like me free extra space.

**Comment by tejas0** [comment]
Found some documentation and ran some debugging sessions with my BLINK setup and have concluded that the query of the sync module:

$sync_units = Invoke-RestMethod -UseBasicParsing $uri -Method Get -Headers $headers

only succeeds in getting the BLINK cameras that are defined to it and are NOT a door bell model nor a mini camera.  I'm not sure if it's because of the call to the "old endpoint" is the reason and there might be a better method for retrieval of the information.  I examined the output of my debugging session and could see it returns all of my outdoor BLINK cameras but not the information for my mini camera in my basement nor my front door camera that I put in place of my door bell with the BLINK door bell camera.

The script does still pull the location paths to these other cameras, but because it never created a sub-folder in which to store their retrieval when it goes through parsing those that need downloading it cannot store them and skips them in the process.  

SOLUTION: 
Create folders (directory names) for any cameras not being downloaded under the sync module name folder created by the script when it goes through the thumbnail process.  It may appeared that there's a problem with the thumbnails and that is a bit misleading.  Forget about them as they don't affect the results.

I currently have one sync module and I've defined it with the name Camera.  That name shows up as the System name.  I've set up outside BLINK cameras with the names Backyard, Driveway, East Side, and Front Yard with them defined to my sync module named Camera.  I have, also, defined to this same sync module additional cameras, a BLINK door bell camera I named Doorbell and a BLINK mini camera I named Basement.

Since the script automatically create the folders with the names of the outdoor cameras but didn't for the other 2 I just manually create folders with those missing with their names under the sync module folder name.  Here's the list:

Sync Module name or "System" name:
U:\Security Cameras\BLINK_Cams\blink\Blink\Camera

Outdoor Cameras found:
U:\Security Cameras\BLINK_Cams\blink\Blink\Camera\Backyard
U:\Security Cameras\BLINK_Cams\blink\Blink\Camera\Driveway
U:\Security Cameras\BLINK_Cams\blink\Blink\Camera\East Side
U:\Security Cameras\BLINK_Cams\blink\Blink\Camera\Front Yard

Other Cameras not found initially:
U:\Security Cameras\BLINK_Cams\blink\Blink\Camera\Basement
U:\Security Cameras\BLINK_Cams\blink\Blink\Camera\Doorbell

Once I did this and ran the script it downloaded automatically all of the associated videos for all of the cameras defined to that sync module/system.  I had specified in the script that I wanted it to use the path as follows:

$saveDirectory = "U:\Security Cameras\BLINK_Cams\blink"

With that in place the script went on to create a further subdirectory structure of
....\Blink\<sync module name>\<camera name>
but as previously detailed failed to included non-Outdoor cameras.  Go figure.  

I hope this helps everyone with this issue.  I haven't worked in quite awhile with REST processes and though I did a little more digging into trying to figure out a way to get the other names this solution of just manually creating the additional subdirectories is the quickest, easiest solution for most people who want to use this script.  Here's my final result (ignore the additional folders not related. Those are from me pulling the local storage thumb drive and syncing it manually as well as some older security cameras).

![image](https://user-images.githubusercontent.com/40218006/222183355-228dcf89-a018-42c0-aef6-f7eeda757036.png)

**Comment by AdrianP22** [comment]
Thank you for looking into it. It's an easy fix! Works great , just added the name of Door Bell to the foulder and they started poping up.

**Comment by stevekenney318** [comment]
> It may appeared that there's a problem with the thumbnails and that is a bit misleading.

@tejas0 - So there is still no fix for downloading thumbnails ? That's currently my only issue and I wish someone could fix it.

**Comment by jg24001** [comment]
Not a script or an ideal fix to your problem, so not ideal but you could
hit the camera icon, have it take a new picture/thumbnail, and then just
screen capture on your phone.

Better than nothing though?

**Comment by stevekenney318** [comment]
> Not a script or an ideal fix to your problem, so not ideal but you could hit the camera icon, have it take a new picture/thumbnail, and then just screen capture on your phone. Better than nothing though?

yeah, I've done that a few times, just looking for the fix to do it automatically like it used to !

**Comment by tejas0** [comment]
Folks, I'm tired and not feeling all that great.  So, I decided not to go through the process of doing a PULL Request and/or Fork.  I'm just posting the information here for your convenience to copy if you so choose.  This one works based on the original one but with some updates.  You no longer need to create your own folders and you will get your thumbnails.

[BlinkCameraDownloads.zip](https://github.com/nayrk/Blink/files/10887001/BlinkCameraDownloads.zip)

As I mentioned, I'm not feeling great and may have fat fingered something in removing my personal information from the code.  So, if you have a problem please let me know.  I'm not normally a PowerShell script coder, but in a previous life I did spend a lot of time doing all things in IT on all shapes and sizes of equipment but am retired now.

**Comment by stevekenney318** [comment]
That sounds great, I will give it a try later on tonight or tomorrow. Thank you very much !

**Comment by stevekenney318** [comment]
@tejas0 - the script ran without a hitch... As far as issue 19 -> https://github.com/nayrk/Blink/issues/19 , I'm thinking there may have been a slight misunderstanding - maybe on my part. With your changes, I am getting the thumbnails for the videos, but what I was looking for are the pictures shown on the home screen of the app for each camera.. .It used to download those every time you took a new one ... Here is an example of one & the name was 'thumbnail_fw_7.101__ry23SsDB_2022_01_29__20_40PM.jpg' 
![thumbnail_fw_7 101__ry23SsDB_2022_01_29__20_40PM](https://user-images.githubusercontent.com/53465149/222919240-279df778-5127-4723-bb9a-c0966cada404.jpg)

a couple of other items I want to ask ... 

1) I had changed my own script to run every 3 hours instead of 30 minutes ... Way back when I did it, I was able to hack something in to the script. Is there a way you could update the script with a variable to do that ? That would help a lot !! 

2) Apparently somehow I had also changed the script to put all the videos in one folder instead of in separate folders... would that still be possible to do with the way you changed the script... I really don't want them in separate folders by camera name... here's an example of what I have 
![20230304_121117_000110_Blink](https://user-images.githubusercontent.com/53465149/222919522-86ec1e17-3674-473a-bc48-4b186f5ae6ed.jpg)

I do appreciate the work you put into it and not having to see those error messages every time !!

Thanks, Steve

**Comment by tejas0** [comment]
I was not able to resolve the original thumbnail issues sought after by some.  That aside, I did add a variable to change the amount of sleep time and another to either put all downloads into a single folder or keep separate folders.  Also, modified the download file names to reflect what they represent.  If they go into a separate folder then that portion of the name is left off since they will obviously be in the folder indicated by the name of the camera.

# Change sleepTimeSeconds to the number desired or leave at 1800 (30 minutes) as the default
# Set up Default Sleep time
$sleepTimeSeconds = 1800
$sleepTimeMinutes = $sleepTimeSeconds/60

# Change whether or not each camera media is stored in its own folder/subdirectory or if all are stored in just one.
# A numeric value of 1 (one) will use 1 folder and the default of 0 (zero) will have each camera have its own folder.
$useSingleFolder = 0

The new script can be found in here:
[BlinkCameraDownloads2.zip](https://github.com/nayrk/Blink/files/10973699/BlinkCameraDownloads2.zip)

**Comment by stevekenney318** [comment]
> ...That aside, I did add a variable to change the amount of sleep time and another to either put all downloads into a single folder or keep separate folders...

This is wonderful !! Thank you so much for adding the timer variable , and the ability to put them all in one folder ... Hopefully the original issue of the thumbnails can be figured out someday !! 

I do have a question (and may have some more after using it) .... I keep all my videos in a folder within my Blink folder called Videos ... If I change  $path = "$saveDirectory\Blink\$networkName\AllCameras" to $path = "$saveDirectory\Blink\Videos" , that should work, correct ?

**Comment by tejas0** [comment]
Yes, it will put the files under that folder but will still create the
folder structure that conforms to the network/system name assigned to each
Sync2 module much as it is doing now.

**Comment by stevekenney318** [comment]
> Yes, it will put the files under that folder but will still create the folder structure that conforms to the network/system name assigned to each Sync2 module much as it is doing now.

Thanks, I did have to go in and change some more.  I didn't give it a chance to make the folder structure. And it seems to be doing what I want now. I appreciate the work you put into it !

![20230315_145037_000240_BlinkCameraDownloads2 - Original ps1 ↔ BlinkCameraDownloads2 ps1 - Visual Studio](https://user-images.githubusercontent.com/53465149/225414021-41446054-d007-4553-88a1-d9e018a2a797.jpg)

![20230315_145058_000241_BlinkCameraDownloads2 - Original ps1 ↔ BlinkCameraDownloads2 ps1 - Visual Studio](https://user-images.githubusercontent.com/53465149/225414017-27b5a69d-5ebc-4d53-bda1-5d23f6ce92d3.jpg)

**Comment by ccclapp** [comment]
Thanks guys, adding the missing doorbell camera folder manually solved this for me
